### PR TITLE
ODC-6779 - Hide sub-catalog(s) in the developer catalog or the entire dev catalog based on customization

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
JIRA story:
https://issues.redhat.com/browse/ODC-6779

Description:
This story is about the customization of developer catalog and sub-catalog based on customization. If all the available sub-catalogs(types) are disabled, complete developer catalog along with the entry points to the catalog will be disabled. 

Screenshots:
----When all the sub-catalogs are disabled - Add page----
Screenshot 2022-09-20 at 12 38 17 PM

----When all the sub-catalogs are disabled - Topology page----
Screenshot 2022-09-20 at 12 38 34 PM

-----Quick start when complete developer catalog is disabled-----
Screenshot 2022-09-20 at 12 38 52 PM

------ All services - when all the sub-catalogs(types) are enabled----
Screenshot 2022-09-20 at 2 18 35 PM

------ All services - when Templates and Helm charts sub-catalogs are disabled----
Screenshot 2022-09-20 at 2 18 44 PM

-----Helm Page when helm chart sub-catalog is disabled-----
Screenshot 2022-09-20 at 12 38 01 PM

---- Operator hub description when operator backed service is disabled or complete developer catalog is disabled---
Screenshot 2022-09-20 at 12 39 37 PM

---- If Event source type is disabled, create event source in action menu is hidden----
Screenshot 2022-09-20 at 1 00 52 PM

-----If user try to reach to catalog by entering URL and if that sub-catalog is disabled, then user will see Page not found--
Screenshot 2022-09-20 at 12 46 20 PM

Test cases:

Screenshot 2022-09-20 at 3 26 05 PM

How to test:

If backend API PR's - https://github.com/openshift/api/pull/1256 & https://github.com/openshift/console-operator/pull/676 are not merged then to add the customization use below method,

Pass options as a string to bridge with -developer-catalog-disabled-types or pass a ConsoleConfig yaml with -config option.

Example:

Specify developer catalog types to bridge with ./bin/bridge -developer-catalog-types '{ "state": "Disabled" ,"disabled": ["HelmChart", "Devfile"] }'.

This indicated HelmChart and Devfile sub-catalogs are disabled. Similarly below are the test scenarios,

If the `state` is `Enabled` and `enabled` list is empty, all the types will be shown.
If the `state` id `Enabled` and `enabled` list is non-empty, only the `enabled` list items will be shown.
If the `state` is `Disabled` and `disabled` list is empty, all the types will be hidden.
If the `state` id `Disabled` and `disabled` list is non-empty, only the `disabled` list items will be hidden.

To disable the complete developer catalog then in `disabled` all the available types
should be listed or make the `disabled` list empty for `state: Disabled`. 

/kind feature

### CODE REVIEW

File: frontend/packages/dev-console/src/actions/providers.ts [hunk 1]
```diff
@@ -63,6 +68,12 @@ const resourceAttributes = (model: K8sModel, namespace: string): AccessReviewRes
   };
 };
 
+export const developerCatalogProvider = (setFeatureFlag: SetFeatureFlag) => {
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/actions/providers.ts:71]:
```suggestion
export const useDeveloperCatalogProvider = (setFeatureFlag: SetFeatureFlag) => {
```

File: frontend/packages/dev-console/src/utils/useAddActionExtensions.ts [hunk 2]
```diff
@@ -32,3 +40,68 @@ export const useAddActionExtensions = (): [ResolvedExtension<AddAction>[], boole
...
+      }
+      if (developerCatalogTypes?.disabled && developerCatalogTypes?.disabled.length === 0) {
+        return false;
+      }
+    }
+  }
+  return true;
+};
+
+export const GetAllSubCatalogsDisabled = (catalogType?: string) => {
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply frontend/packages/dev-console/src/utils/useAddActionExtensions.ts:63]: you are not passing any value to catalog type parameter while calling `GetAllSubCatalogsDisabled` anywhere

File: frontend/packages/dev-console/src/utils/useAddActionExtensions.ts [hunk 3]
```diff
@@ -32,3 +40,68 @@ export const useAddActionExtensions = (): [ResolvedExtension<AddAction>[], boole
 
   return [allAddActionExtensions, resolved, allAddActionsDisabled];
 };
+
+export const getEnabledSubCatalogTypes = (catalogType: string): boolean => {
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/utils/useAddActionExtensions.ts:44]:
```suggestion
export const isSubcatalogTypeEnabled = (catalogType: string): boolean => {
```

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/actions.ts [hunk 4]
```diff
@@ -126,6 +126,8 @@ export type Action = {
   insertAfter?: string | string[];
   /** Describes the access check to perform. */
   accessReview?: AccessReviewResourceAttributes;
+  /** check if sub-catalog type is enabled or not */
+  isEnabled?: boolean;
```
Comment (jerolimov) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/actions.ts:130]: This is a generic "Action" and should not contain specific attributes for "sub-catalog". We should also just do not add actions instead of setting isEnabled=false. Just from reading the action extension it would be unclear what the difference is between "disabled" and "isEnabled". I hope we can find a solution without adding this property here.

```suggestion
```

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/catalog.ts [hunk 5]
```diff
@@ -17,6 +17,8 @@ export type CatalogItemType = ExtensionDeclaration<
     filters?: CatalogItemAttribute[];
     /** Custom groupings specific to the catalog item. */
     groupings?: CatalogItemAttribute[];
+    /** Indicated the sub-catalog is enabled or not */
+    enabledType?: boolean;
```
Comment (jerolimov) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/catalog.ts:21]: Is this really needed? Isn't it possible to just skip these entries?

File: frontend/packages/dev-console/src/actions/add-resources.tsx [hunk 6]
```diff
@@ -60,7 +61,14 @@ export const resolvedURLWithParams = (
...
 
 export const AddActions: { [name: string]: ActionFactory } = {
-  FromGit: (namespace, application, contextSource, path, accessReviewDisabled) => ({
+  FromGit: (
+    namespace,
+    application,
+    contextSource,
+    path,
+    accessReviewDisabled,
+    isActionTypeEnabled,
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/actions/add-resources.tsx:70]: Instead of adding more and more parameters to this "helpers", could we just skip calling `FromGit` (and all others) if this parameter is false?

File: frontend/packages/dev-console/src/utils/useAddActionExtensions.ts (hunk 3 above)
Comment (jerolimov) [reply frontend/packages/dev-console/src/utils/useAddActionExtensions.ts:44]: I would recommend `isCatalogTypeEnabled` as the extension is called `CatalogType`

File: frontend/packages/dev-console/src/utils/useAddActionExtensions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jerolimov) [reply frontend/packages/dev-console/src/utils/useAddActionExtensions.ts:63]: And please use a lower case `get...` as we do anywhere else for function names.

File: frontend/packages/helm-plugin/src/components/list-page/HelmTabbedPage.tsx [hunk 7]
```diff
@@ -62,7 +63,7 @@ export const PageContents: React.FC<HelmTabbedPageProps> = (props) => {
 
   const menuActions: MenuActions = {
     helmRelease: {
-      label: t('helm-plugin~Helm Release'),
+      label: isHelmVisible ? t('helm-plugin~Helm Release') : null,
       onSelection: () => `/catalog/ns/${namespace}?catalogType=HelmChart`,
     },
```
Comment (jerolimov) [inline frontend/packages/helm-plugin/src/components/list-page/HelmTabbedPage.tsx:68]: If the flag is false and we don't want to add an `Action`, we should not add it completely instead of setting the label to null.

File: frontend/packages/knative-plugin/src/actions/add-event-sink.tsx [hunk 8]
```diff
@@ -12,6 +12,7 @@ export const AddEventSinkAction = (
   application?: string,
   contextSource?: string,
   path?: string,
+  isEventSinkTypeEnabled?: boolean,
```
Comment (jerolimov) [inline frontend/packages/knative-plugin/src/actions/add-event-sink.tsx:15]: We should not call this function if the `Action` is not enabled "for any reason"

**Review by jerolimov — CHANGES_REQUESTED** [review]
This was just a high level review but we should not update the extensions if it's not needed.

PTAL on my comments below:

File: frontend/packages/dev-console/src/utils/useAddActionExtensions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lokanandaprabhu) [reply frontend/packages/dev-console/src/utils/useAddActionExtensions.ts:63]: If we want to use this function for particular catalogType, then we can use this directly by passing catalogtype.. So I kept catalogType as input 

If I use lowercase as function name.. I was getting below error since I am calling another hook inside that function.. I did some research and the suggestion was to keep starting letter as uppercase.. any feedback on this will be helpful

```
74:32  error  React Hook "useResolvedExtensions" is called in function "getAllSubCatalogsDisabled" that is neither a React function component nor a custom React Hook function  react-hooks/rules-of-hooks
  75:5   error  React Hook "React.useCallback" is called in function "getAllSubCatalogsDisabled" that is neither a React function component nor a custom React Hook function      react-hooks/rules-of-hooks
```

File: frontend/packages/helm-plugin/src/components/list-page/HelmTabbedPage.tsx (hunk 7 above)
Comment (lokanandaprabhu) [reply frontend/packages/helm-plugin/src/components/list-page/HelmTabbedPage.tsx:68]: If the label is null then that action will not be added.

File: frontend/packages/dev-console/src/utils/useAddActionExtensions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lokanandaprabhu) [reply frontend/packages/dev-console/src/utils/useAddActionExtensions.ts:63]: I will remove `catalogType`  from input parameter

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/actions.ts (hunk 4 above)
Comment (lokanandaprabhu) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/actions.ts:130]: The existing `disabled` property will not hide the option, it will just disable it and user can see the action is disabled. So I added isEnabled to hide the option. I will check how can I achieve this without `isEnabled`.

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/catalog.ts (hunk 5 above)
Comment (lokanandaprabhu) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/catalog.ts:21]: I will remove `enabledType`

File: frontend/packages/dev-console/src/utils/__tests__/isSubCatalogTypeEnabled.spec.ts [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+const useResolvedExtensionsMock = useResolvedExtensions as jest.Mock;
+useResolvedExtensionsMock.mockReturnValue([mockExtensions, true]);
+
+beforeEach(() => {
+  delete window.SERVER_FLAGS.developerCatalogTypes;
+});
+
+describe('isSubCatalogTypeEnabled - get sub-catalog status', () => {
+  it('if customization is not there', () => {
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/utils/__tests__/isSubCatalogTypeEnabled.spec.ts:22]:
```suggestion
  it('should show HelmChart catalog type as enabled when dev catalog types are not configured', () => {
```

File: frontend/packages/dev-console/src/utils/__tests__/isSubCatalogTypeEnabled.spec.ts [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+    const isEnabled = isCatalogTypeEnabled('HelmChart');
+    expect(isEnabled).toBe(true);
+  });
+  it('if none of the sub-catalogs enabled', () => {
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/utils/__tests__/isSubCatalogTypeEnabled.spec.ts:26]: update the rest in a similar way

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+                Allows cluster admins to selectively disable a sub-catalog and all links to the specific sub-catalog or disable the complete developer catalog
+
+        Background:
+            Given user has created or selected namespace "aut-developer-catalog"
+
+ @regression @manual
+        Scenario: When all the sub-catalogs are disabled: DC-01-TC01
+            Given user is at cluster YAML of "operator.openshift.io/v1" console
+             When user adds the "Add sub-catalog types" code snippet under spec.customization.developerCatalog.types
+              And change the state to "Disabled" and remove enabled attribute
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature:12]:
```suggestion
              And user changes the state to "Disabled" and removes enabled attribute
```

File: frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+              And user clicks on Save button
+             Then user will not see "Developer Catalog" and all the sub-catalogs in Add page and Topology page
+              And user will not get any link to catalog page 
+
+  @regression @manual
+        Scenario: When all the sub-catalogs are enabled: DC-01-TC02
+            Given user is at cluster YAML of "operator.openshift.io/v1" console
+             When user adds the "Add sub-catalog types" code snippet under spec.customization.developerCatalog.types
+              And keep the state as "Enabled" and remove enabled attribute
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature:21]:
```suggestion
              And user keeps the state as "Enabled" and removes enabled attribute
```

Comment (lokanandaprabhu) [reply frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature]: Updated the test cases accordingly.

File: frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+            Given user has created or selected namespace "aut-developer-catalog"
+
+ @regression @manual
+        Scenario: When all the sub-catalogs are disabled: DC-01-TC01
+            Given user is at cluster YAML of "operator.openshift.io/v1" console
+             When user adds the "Add sub-catalog types" code snippet under spec.customization.developerCatalog.types
+              And user changes the state to "Disabled" and removes enabled attribute
+              And user clicks on Save button
+             Then user will not see "Developer Catalog" and all the sub-catalogs in Add page and Topology page
+              And user will not get any link to catalog page
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature:15]: This should be well defined i.e. define from where the catalog link will be disappeared

Comment (lokanandaprabhu) [reply frontend/packages/dev-console/integration-tests/features/customization/customization-of-catalog.feature]: Hi @sanketpathak , added the details.

**Comment by sanketpathak** [comment]
@lokanandaprabhu While testing your pr on cluster-bot, I saw this scenario:

- I enabled the dev catalog for Helm charts only
- When I switch to topology after a hard refresh
- I can see the Database card, is this an expected behavior
[Screencast from 10-12-2022 07:21:44 PM.webm](https://user-images.githubusercontent.com/10252230/195362173-3e302867-20a4-4be5-8929-5d6834284c3f.webm)

**Comment by lokanandaprabhu** [comment]
> @lokanandaprabhu While testing your pr on cluster-bot, I saw this scenario:
> 
> * I enabled the dev catalog for Helm charts only
> * When I switch to topology after a hard refresh
> * I can see the Database card, is this an expected behavior
>   [Screencast from 10-12-2022 07:21:44 PM.webm](https://user-images.githubusercontent.com/10252230/195362173-3e302867-20a4-4be5-8929-5d6834284c3f.webm)

Yes, Database will be enabled when atleast one sub-catalog is enabled. Database will be disabled only when the complete developer catalog is disabled.

**Comment by sanketpathak** [comment]
Verified on cluster-bot
[Screencast from 10-13-2022 03:50:25 PM.webm](https://user-images.githubusercontent.com/10252230/195572822-293ab4bb-056a-49cd-a0a4-e020d18df88e.webm)

/label qe-approved

**Comment by vikram-raj** [comment]
- Event source option is present in the connector context menu which leads to a 404 page. Even after disabling the Event Source catalog. See the GIF
Peek 2022-10-13 18-32

- Event Sink option is also present in the connector context menu. And in the action menu of the broker which leads to a 404 page.
image

image


- On disabling the complete catalogs. Should we hide the `Create applications using samples` from the `Getting started resources ` card as `View all samples` links leads to an empty Sample catalog page? 
image

**Comment by lokanandaprabhu** [comment]
Updated the PR. PTAL. Thanks.

**Comment by debsmita1** [comment]
The entire dev-catalog is disabled, but I can see `Samples` option when I click on topology graph
https://user-images.githubusercontent.com/22490998/195789490-c2df4321-a88d-4c1b-82c1-ac5e4a05cccc.mov

**Comment by lokanandaprabhu** [comment]
> The entire dev-catalog is disabled, but I can see `Samples` option when I click on topology graph https://user-images.githubusercontent.com/22490998/195789490-c2df4321-a88d-4c1b-82c1-ac5e4a05cccc.mov

Updated the PR.

File: frontend/packages/dev-console/src/actions/providers.ts [hunk 14]
```diff
@@ -88,81 +103,127 @@ export const useTopologyGraphActionProvider: TopologyActionProvider = ({
...
+          undefined,
+        ),
+      );
+    }
+    actionsWithoutSourceRef.push(
+      AddActions.UploadJarFile(
+        namespace,
+        undefined,
+        undefined,
+        'add-to-project',
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/actions/providers.ts:206]: nit: better to add this string `add-to-project` in a const.

**Review by vikram-raj — COMMENTED** [review]
Thanks @lokanandaprabhu . This feature works as expected. Just added a couple of nits. PTAL if it makes sense.

File: frontend/packages/console-shared/src/components/catalog/service/CatalogServiceProvider.tsx [hunk 15]
```diff
@@ -47,7 +48,7 @@ const CatalogServiceProvider: React.FC<CatalogServiceProviderProps> = ({
     catalogBadgeProviderExtensions,
     extensionsResolved,
   ] = useCatalogExtensions(catalogId, catalogType);
-
+  const [disabledSubCatalogs] = useGetAllDisabledSubCatalogs();
```
Comment (jerolimov) [inline frontend/packages/console-shared/src/components/catalog/service/CatalogServiceProvider.tsx:51]: Isn't it easier to filter the items directly in `useCatalogExtensions` so that it returns already just an array of enabled `catalogTypeExtensions`?

File: frontend/packages/dev-console/src/const.ts [hunk 16]
```diff
@@ -26,3 +26,9 @@ export const LAST_RESOURCE_TYPE_STORAGE_KEY = `devconsole.last.resource-type`;
 
 export const NAME_LABEL = 'app.kubernetes.io/name';
 export const INSTANCE_LABEL = 'app.kubernetes.io/instance';
+export const FLAG_DEVELOPER_CATALOG = 'DEVELOPER_CATALOG';
+export const FLAG_OPERATOR_BACKED_SERVICE_CATALOG_TYPE = 'OPERATOR_BACKED_SERVICE_CATALOG_TYPE';
+export const FLAG_SAMPLE_CATALOG_TYPE = 'SAMPLE_CATALOG_TYPE';
+export const OPERATOR_BACKED_SERVICE_SUB_CATALOG = 'OperatorBackedService';
+export const SAMPLE_SUB_CATALOG = 'Sample';
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/const.ts]: Aren't these all catalog types? Why are some the suffix `_CATALOG_TYPE` and some `_SUB_CATALOG`?

These are catalog IDs, right? Can we add this to the names?

```tsx
export const OPERATOR_BACKED_SERVICE_CATALOG_TYPE_ID = 'OperatorBackedService';
export const SAMPLE_CATALOG_TYPE_ID = 'Sample';
```

File: frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogTypeSelector.tsx [hunk 17]
```diff
@@ -49,7 +50,7 @@ const CatalogTypeSelector: React.FC<CatalogTypeSelectorProps> = ({
             search: `?${queryParams.toString()}`,
           };
 
-          return typeCount > 0 ? (
+          return typeCount > 0 && !disabledSubCatalogs?.includes(value) ? (
```
Comment (jerolimov) [inline frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogTypeSelector.tsx:53]: Instead of keeping the `catalogTypes` prop with all items from `CatalogController` to `CatalogView` and hiding the values here, could we filter these already in `CatalogController`?

The code there looks like this and should be easy to filter, or?

```tsx
  const catalogTypes: CatalogType[] = React.useMemo(() => {
    const types = catalogExtensions.map((extension) => ({
      label: extension.properties.title,
      value: extension.properties.type,
      description: extension.properties.typeDescription,
    }));

    return _.sortBy(types, ({ label }) => label.toLowerCase());
  }, [catalogExtensions]);
```

(untested)

File: frontend/packages/dev-console/src/utils/catalog-utils.ts [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+        return [developerCatalogTypes?.disabled, catalogTypeExtensions];
+      }
+      return [catalogTypeExtensions, catalogTypeExtensions];
+    }
+  }
+  return [[], catalogTypeExtensions];
+};
+
+export const useIsDeveloperCatalogEnabled = (): boolean => {
+  const [disabledSubCatalogs, catalogExtensionsArray] = useGetAllDisabledSubCatalogs();
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/utils/catalog-utils.ts:59]: `useGetAllDisabledSubCatalogs` returns two different arrays just for this case here, right?

Doesn't it make sense that `useIsDeveloperCatalogEnabled` calls `const [catalogExtensionsArray] = useResolvedExtensions<CatalogItemType>(isCatalogItemType);` itself to deterinate if all catalogs are enabled/disabled?

Then `useGetAllDisabledSubCatalogs` can just return the disabled array instead of two arrays. Doesn't this make it easier to understand what it will return based on the function name? And isn't useGetDisabledCatalogTypes more exact what it returns?

File: frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogTypeSelector.tsx [hunk 19]
```diff
@@ -18,7 +18,6 @@ const CatalogTypeSelector: React.FC<CatalogTypeSelectorProps> = ({
 }) => {
   const { t } = useTranslation();
   const { pathname, search } = useLocation();
-
```
Comment (jerolimov) [inline frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogTypeSelector.tsx:21]: Unnecessary change

File: frontend/packages/kubevirt-plugin/src/components/create-vm/dev-console/dev-console-create-vm-form.tsx [hunk 20]
```diff
@@ -13,6 +13,7 @@ import { isEmpty } from 'lodash';
 import Helmet from 'react-helmet';
 import { Trans, useTranslation } from 'react-i18next';
 import { RouteComponentProps } from 'react-router';
+import { isCatalogTypeEnabled } from '@console/dev-console/src/utils/catalog-utils';
```
Comment (jerolimov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/dev-console/dev-console-create-vm-form.tsx:16]: We should not add a dependency from the `kubevirt-plugin` to `dev-console`.

Please move the utils from `@console/dev-console/src/utils/catalog-utils` to `@console/shared/src/components/catalog/utils/catalog-utils`.

File: frontend/packages/dev-console/src/utils/catalog-utils.ts [hunk 21]
```diff
@@ -0,0 +1,70 @@
+import {
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/utils/catalog-utils.ts:1]: We should not add a dependency from other plugins to `dev-console`.

Please move this utils and their tests into `packages/console-shared/src/components/catalog/utils/catalog-utils.tsx`

**Review by jerolimov — CHANGES_REQUESTED** [review]
This PR needs a rebase and I would like that you move the utils as mentioned below.

**Comment by jerolimov** [comment]
Following our no-FF process propagate the PX and Docs ACK from [epic ODC-5897](https://issues.redhat.com/browse/ODC-5897)
/label px-approved
/label docs-approved

I add also approve here so that @debsmita1 can do the final review when the files are moved (https://github.com/openshift/console/pull/12067#pullrequestreview-1143240309) and the PR is rebased.
/approve

**Comment by lokanandaprabhu** [comment]
Hi @jerolimov / @debsmita1 ,

I have updated the PR based on PR comments. PTAL. Thanks.

**Review by jerolimov — APPROVED** [review]
Tested the latest version on a cluster bot instance and it works fine. :+1: 

/lgtm
